### PR TITLE
introduction to using OAuth2 setup and refresh

### PR DESCRIPTION
**Pull request by richardbahari**
This is related to ch17081, introducing the v2 flow using OAuth2 aspect to the SDK. From v2 onwards, OAuth2 tokens are going to be required. 
In the context.properties, the testurl can now be replaced with the new v2 flow using an API Gateway url. There are also properties related to the oauth2 tokens.

### CODE REVIEW

File: overledger-sdk-example/src/main/java/network/quant/SDKOauthSample.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+            System.out.println("Result after refreshing: " + sdk.refreshAccessToken());
+
+        } catch (Exception e) {
+            System.out.println("Some error happened in the demo, make sure you have the config file in the src/main/resources set and the oauth2 values set in the context.properties");
+            e.printStackTrace();
+        }
+
+
+    }
```
Comment (kashif-quant) [inline overledger-sdk-example/src/main/java/network/quant/SDKOauthSample.java:26]: Minor points: I think the grammar is incorrect here, "Some error occurred..." seems more appropriate. Also you have a few extra blank lines that can be removed.

Comment (richardbahari) [reply overledger-sdk-example/src/main/java/network/quant/SDKOauthSample.java:22]: done

File: overledger-sdk-api/src/main/java/network/quant/api/Client.java [hunk 2]
```diff
@@ -127,4 +128,4 @@
     List<EventSubscribeResponse> postUnsubscribeEvent(DltTransaction dltTransaction);
     ContractQueryResponseDto smartContractQuery(DltTransaction dltTransaction, String dlt);
 
-}
+}
```
Comment (alexchiriac) [inline overledger-sdk-api/src/main/java/network/quant/api/Client.java:131]: Missing newline :)

File: overledger-sdk-api/src/main/java/network/quant/api/OverledgerSDK.java [hunk 3]
```diff
@@ -188,4 +193,4 @@
     List<EventSubscribeResponse> eventSubscribe(DltTransaction dltTransaction);
     List<EventSubscribeResponse> eventUnsubscribe(DltTransaction dltTransaction);
     ContractQueryResponseDto smartContractQuery(DltTransaction contractQueryRequestDto, String dlt);
-}
+}
```
Comment (alexchiriac) [inline overledger-sdk-api/src/main/java/network/quant/api/OverledgerSDK.java:196]: Missing newline :)

File: overledger-sdk-api/src/main/java/network/quant/util/Oauth2RefreshResponse.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    @Override
+    public String toString() {
+        return "Oauth2RefreshResponse{" +
+                "id_token='" + id_token + '\'' +
+                ", access_token='" + access_token + '\'' +
+                ", token_type='" + token_type + '\'' +
+                ", expires_in='" + expires_in + '\'' +
+                '}';
+    }
+}
```
Comment (alexchiriac) [inline overledger-sdk-api/src/main/java/network/quant/util/Oauth2RefreshResponse.java:65]: Missing newline :)

File: overledger-sdk-essential/src/main/java/network/quant/essential/OverledgerClient.java [hunk 5]
```diff
@@ -502,10 +546,10 @@ public Block searchBlock(String dlt, String blockhash, Class<Block> responseClas
...
-            I = new OverledgerClient();
+        if (null == client) {
+            client = new OverledgerClient();
         }
-        return I;
+        return client;
     }
 
-}
+}
```
Comment (alexchiriac) [inline overledger-sdk-essential/src/main/java/network/quant/essential/OverledgerClient.java:555]: Missing newline :)

File: overledger-sdk-example/src/main/java/network/quant/SDKOauthSample.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            OverledgerSDK sdk = DefaultOverledgerSDK.newInstance();
+
+            System.out.println("Result after refreshing: " + sdk.refreshAccessToken());
+
+        } catch (Exception e) {
+            System.out.println("Some error occurred in the demo, make sure you have the config file in the src/main/resources set and the oauth2 values set in the context.properties");
+            e.printStackTrace();
+        }
+    }
+}
```
Comment (alexchiriac) [inline overledger-sdk-example/src/main/java/network/quant/SDKOauthSample.java:23]: Missing newline :)

**Review by alexchiriac — APPROVED** [review]
LGTM! Left some comments about missing newlines at the end of files

File: overledger-sdk-example/src/main/java/network/quant/SDKOauthSample.java (hunk 6 above)
Comment (richardbahari) [reply overledger-sdk-example/src/main/java/network/quant/SDKOauthSample.java]: done

File: overledger-sdk-essential/src/main/java/network/quant/essential/OverledgerClient.java (hunk 5 above)
Comment (richardbahari) [reply overledger-sdk-essential/src/main/java/network/quant/essential/OverledgerClient.java]: done

File: overledger-sdk-api/src/main/java/network/quant/util/Oauth2RefreshResponse.java (hunk 4 above)
Comment (richardbahari) [reply overledger-sdk-api/src/main/java/network/quant/util/Oauth2RefreshResponse.java]: done

File: overledger-sdk-api/src/main/java/network/quant/api/OverledgerSDK.java (hunk 3 above)
Comment (richardbahari) [reply overledger-sdk-api/src/main/java/network/quant/api/OverledgerSDK.java]: done

File: overledger-sdk-api/src/main/java/network/quant/api/Client.java (hunk 2 above)
Comment (richardbahari) [reply overledger-sdk-api/src/main/java/network/quant/api/Client.java]: done

**Review by theoQuant — APPROVED** [review]
Looks good

**Review by kashif-quant — APPROVED** [review]

**Review by mahsa-bytt — APPROVED** [review]

**Pull request merged**